### DISCUSSION TITLE
Spotify Plugin

### DISCUSSION THREAD

**Discussion by bji219** [danny-avila/LibreChat#542] [Feature Requests & Suggestions]
The detailed instructions in "Create your own" plugin markdown are very detailed and I am sure a more experienced programmer would be able to implement this much easier than myself! The gist of the idea is to create a custom plugin using the Spotify API to create playlists or find unique/fun songs based on natural language prompts! Let me know if anyone is interested in collaborating!

**Comment by danny-avila** [comment]
You may find this useful! They use the python langchain library but I believe most of the features are available in JS too.

https://jonathansoma.com/words/custom-execution-chain.html

Also take a look at structured tools, which work with the functions agent:
https://github.com/danny-avila/LibreChat/blob/main/api/app/langchain/tools/structured/StableDiffusion.js

**Reply by bji219** [reply]
This is a great article, thanks for sharing!

**Comment by fuegovic** [comment]
@bji219 Great idea, I almost started to work on something like that but got busy and forgot about it
